src/aof.c

```diff
@@ -572,6 +572,16 @@ int writeAofManifestFile(sds buf) {
             tmp_am_name, am_name, strerror(errno));
 
         ret = C_ERR;
+        goto cleanup;
+    }
+
+    /* Also sync the AOF directory as new AOF files may be added in the directory */
+    if (fsyncFileDir(am_filepath) == -1) {
+        serverLog(LL_WARNING, "Fail to fsync AOF directory %s: %s.",
+            am_filepath, strerror(errno));
+
+        ret = C_ERR;
```

Comment:
I read the [man page](https://man7.org/linux/man-pages/man2/fdatasync.2.html), the error condition, like bad fd, disk was exhausted, there's error in previous `write` but it's reported by `fsync`, seems unlikely to happen to directory. If `fsync` fails, we can consider the data is in page cache and cannot persist to disk at the moment. If the error is recoverable like insufficient disk, the data will flush to disk eventually, otherwise the data may be lost. All in all, the error seems unlikely to happen to directory.

From [Leveldb](https://github.com/google/leveldb/blob/d019e3605f222ebc5a3a2484a2cb29db537551dd/util/env_posix.cc#L342) and [Rocksdb](https://github.com/facebook/rocksdb/blob/e55018a8ce0b83e3f1dbe17d0632062924e119ad/file/filename.cc#L408), they both regard the dir sync failure as a normal error(the same as write, rename and other file operations),  so maybe we should also treat them conservatively though the failure may never occur at all.

Reply:
I tend to agree, fsync error here is like a write error.